### ISSUE TITLE
Artist Image (again)

### ISSUE THREAD

**Issue by aburchio** [navidrome/navidrome#2210]
I know there have been many threads about this, but I can't find a similar situation.
I have spotify integration setup and working, I think it works because in the logs I can find the URLs to the images, and are valid ones. But in the brosers window it will always show the default gray star. 
I switched LOG level to debug, and eventually to TRACE because I could not find relevant information.

 - Navidrome version: Version: 0.49.3-SNAPSHOT (5fc4076a)
 - Browser and version: Firefox 110.0 (64 bit) and MS Edge on Windows10
 - Operating System: docker container hosted on Debian 11

Here is the logs when loading an Artist page inside Navidrome:

time="2023-02-24T17:13:55Z" level=debug msg="Got Images" agent=spotify artist="Paul Gilbert" elapsed=232.7ms images="[{https://i.scdn.co/image/ab6761610000e5eb22496746b1e2b9d73509b118 640} {https://i.scdn.co/image/ab6761610000517422496746b1e2b9d73509b118 320} {https://i.scdn.co/image/ab6761610000f17822496746b1e2b9d73509b118 160}]" requestId=be31b622276a/6PCynjzn0a-000018 username=admin

time="2023-02-24T17:13:56Z" level=trace msg="Putting a keepalive event on client's queue" client="a73aa665-7716-4621-b429-dd558caf3d82 (admin - 192.168.89.174:51207 -  - Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0)" event="{3018 keepAlive {\"ts\":1677258836} 0xc000040040}"

time="2023-02-24T17:13:56Z" level=trace msg="Putting a keepalive event on client's queue" client="0d6539e1-31ae-4b55-98f5-22487d7d2a7e (admin - 192.168.89.174:51206 -  - Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0)" event="{3018 keepAlive {\"ts\":1677258836} 0xc000040040}"

time="2023-02-24T17:13:56Z" level=trace msg="Sending event to client" client="a73aa665-7716-4621-b429-dd558caf3d82 (admin - 192.168.89.174:51207 -  - Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0)" 
event="{3018 keepAlive {\"ts\":1677258836} 0xc000040040}" requestId=be31b622276a/6PCynjzn0a-000003 username=admin

time="2023-02-24T17:13:56Z" level=trace msg="Sending event to client" client="0d6539e1-31ae-4b55-98f5-22487d7d2a7e (admin - 192.168.89.174:51206 -  - Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/110.0)" event="{3018 keepAlive {\"ts\":1677258836} 0xc000040040}" requestId=be31b622276a/6PCynjzn0a-000004 username=admin

what am I missing?

**Comment by deluan** [comment]
Here's a way to troubleshoot artwork resolution: https://github.com/navidrome/navidrome/issues/2190#issuecomment-1434018334

Let me know if this helps figuring this out.

**Comment by aburchio** [comment]
I think this can be the relevant part:

time="2023-03-10T22:47:47Z" level=debug msg="Got Images" agent=spotify artist="Eric Clapton" elapsed=2ms images="[{https://i.scdn.co/image/ab6772690000c46ca60e8f215103f6841d8a83f0 1000} {https://i.scdn.co/image/ab6772690000dd22a60e8f215103f6841d8a83f0 640} {https://i.scdn.co/image/ab6772690000bac3a60e8f215103f6841d8a83f0 200} {https://i.scdn.co/image/ab67726900008f74a60e8f215103f6841d8a83f0 64}]" requestId=be31b622276a/XdsoHPqFWL-000078 username=admin

time="2023-03-10T22:47:52Z" level=trace msg=**"Failed trying to extract artwork"** artID=ar-ecab6f2c92849270334731399109a6c0_0 elapsed=5s error="Get \"https://i.scdn.co/image/ab6772690000dd22a60e8f215103f6841d8a83f0\": **context deadline exceeded (Client.Timeout exceeded while awaiting headers)"** requestId=be31b622276a/XdsoHPqFWL-000078 source=fromArtistExternalSource username=admin

from what I understand, it is a timeout while trying to get the image from spotify server, but I don't know what can be the reason. Opening the link from the logs in a browser loads the image immediately.

**Comment by deluan** [comment]
> from what I understand, it is a timeout while trying to get the image from spotify server, but I don't know what can be the reason. Opening the link from the logs in a browser loads the image immediately.

Yes, that's what is happening. Maybe for some reason the connection from inside the container is blocked? Check if you can access Spotify from inside the container. If you are using docker compose:
```shell
$ docker compose exec navidrome /bin/sh
/app $ cd /tmp
/tmp $ ls -la
total 8
drwxrwxrwt    1 root     root          4096 Mar 10 23:35 .
drwxr-xr-x    1 root     root          4096 Mar 10 09:00 ..
/tmp $ wget https://i.scdn.co/image/ab6772690000dd22a60e8f215103f6841d8a83f0
Connecting to i.scdn.co (146.75.118.248:443)
saving to 'ab6772690000dd22a60e8f215103f6841d8a83f0'
ab6772690000dd22a60e 100% |*******************************************************************************| 96297  0:00:00 ETA
'ab6772690000dd22a60e8f215103f6841d8a83f0' saved
/tmp $ ls -la
total 104
drwxrwxrwt    1 root     root          4096 Mar 10 23:36 .
drwxr-xr-x    1 root     root          4096 Mar 10 09:00 ..
-rw-r--r--    1 1000     1000         96297 Mar 10 23:36 ab6772690000dd22a60e8f215103f6841d8a83f0
/tmp $
```
